payjoin-cli/src/app/v2/mod.rs

```diff
@@ -286,27 +286,75 @@ impl AppTrait for App {
         }
 
         let mut tasks = Vec::new();
+        let mut skipped_sessions = 0;
 
+        // Process receiver sessions
         for session_id in recv_session_ids {
             let self_clone = self.clone();
-            let recv_persister = ReceiverPersister::from_id(self.db.clone(), session_id);
-            let receiver_state = replay_receiver_event_log(&recv_persister)
-                .map_err(|e| anyhow!("Failed to replay receiver event log: {:?}", e))?
-                .0;
-            tasks.push(tokio::spawn(async move {
-                self_clone.process_receiver_session(receiver_state, &recv_persister).await
-            }));
+            let recv_persister = ReceiverPersister::from_id(self.db.clone(), session_id.clone());
+            match replay_receiver_event_log(&recv_persister) {
+                Ok((receiver_state, _)) => {
+                    tasks.push(tokio::spawn(async move {
+                        self_clone.process_receiver_session(receiver_state, &recv_persister).await
+                    }));
+                }
+                Err(e) => {
+                    let error_msg = e.to_string();
+                    if error_msg.contains("expired") || error_msg.contains("Expired") {
```

Comment:
Are we not able to match on the internal error ?
https://github.com/payjoin/rust-payjoin/blob/54118485ead1ceda8a0219927fe05300e2d1cf49/payjoin/src/core/error.rs#L80

Reply:
No , as it is pub(crate), i'll have to change it to pub  to be able to match on it. This also apply to the Time struct. am i wrong abou this ?